### PR TITLE
fix(DB/IcecrownCitadel): Fix Rotface resetting behaviour

### PR DESCRIPTION
**Pull request by 0x7FFF**
Add "Hard Reset" Flag to Rotface (similar to Festergut)

## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [X] Database (SAI, creatures, etc).

## Issues Addressed:

- None in the issues tab

Previously, w/o hard reset flag, you could fast reset Rotface which leads to following bugs:
- Gate is not closing after re-pulling (you can verify this by aggroing - feign death - aggroing again)
- Previous slime pools are still present, while the new one are being poured (same as previous bug, just wait till they appear)

## SOURCE:

The changes have been validated through:
- [X] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [X] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [X] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. .go xyz 4446.165 3080.91 360.523 631
2. Pull Rotface and reset him.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Nyeriah** [comment]
Please remove the select query and edit the update query so it doesn’t include database name. Thanks for your contribution!

**Comment by elthehablo** [comment]
For clarification: queries in world db directories will by default use acore world, and similarly for auth and characters when executed on server startup

File: data/sql/updates/pending_db_world/rev_1701079229296730800.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+-- Rotface
+UPDATE `acore_world`.`creature_template` SET `flags_extra`=3221225473 WHERE  `entry`=36627;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1701079229296730800.sql:2]: dont add flags like this. use the add operator. see https://www.azerothcore.org/wiki/sql-standards#flags-amp-bits

**Review by Kitzunu — DISMISSED** [review]

**Review by Nyeriah — APPROVED** [review]

**Comment by Nyeriah** [comment]
@0x7FFF Thanks for the PR 👍

**Pull request merged**